tripal_chado/tests/src/Functional/Plugin/GFF3ImporterTest.php

```diff
@@ -0,0 +1,1333 @@
+
+    // Do checks on the featureprop table as well. Ensures the bio type value got added
+    $results = $chado->query("SELECT * FROM {1:featureprop} 
+      WHERE feature_id = :feature_id AND value LIKE :value;", [
+      ':feature_id' => $gene_feature_id,
+      ':value' => 'protein_coding'
+    ]);
+    $results_object = $results->fetchObject();
+    $this->assertEquals($results_object->value, 'protein_coding');
+    unset($results);
+    unset($results_object);
+
+
+    // Ensures the GAP value got added
+    $results = $chado->query("SELECT * FROM {1:featureprop} 
+      WHERE feature_id = :feature_id AND value LIKE :value;", [
+      ':feature_id' => $gene_feature_id,
+      ':value' => 'test_gap_1'
+    ]);
+    $results_object = $results->fetchObject();
+    $this->assertEquals($results_object->value, 'test_gap_1');
+    unset($results);
+    unset($results_object);
+
+    // Ensures the NOTE value got added
+    $results = $chado->query("SELECT * FROM {1:featureprop} 
+      WHERE feature_id = :feature_id AND value LIKE :value;", [
+      ':feature_id' => $gene_feature_id,
+      ':value' => 'test_gene_001_note'
+    ]);
+    $results_object = $results->fetchObject();
+    $this->assertEquals($results_object->value, 'test_gene_001_note');
+    unset($results);
+    unset($results_object);
+
+    // TODO: To complete
+    /**
+     * Run the GFF loader on gff_duplicate_ids.gff for testing.
+     *
+     * This tests whether the GFF loader detects duplicate IDs which makes a 
+     * GFF file invalid since IDs should be unique. The GFF loader should throw 
+     * and exception which this test checks for
+     */    
+    $gff3_importer = $importer_manager->createInstance('chado_gff3_loader');
+    $run_args = [
+      'files' => [
+        0 => [
+          'file_path' => __DIR__ . '/../../../fixtures/gff3_loader/gff_duplicate_ids.gff'
+        ]
+      ],
+      'schema_name' => $schema_name,
+      'analysis_id' => $analysis_id,
+      'organism_id' => $organism_id,
+      'use_transaction' => 1,
+      'add_only' => 0,
+      'update' => 1,
+      'create_organism' => 0,
+      'create_target' => 0,
+      // regexps for mRNA and protein.
+      're_mrna' => NULL,
+      're_protein' => NULL,
+      // optional
+      'target_organism_id' => NULL,
+      'target_type' => NULL,
+      'start_line' => NULL,
+      'line_number' => NULL, // Previous error without this
+      'landmark_type' => 'supercontig',
+      'alt_id_attr' => NULL,
+      'skip_protein' => NULL,
+    ];
+
+    $file_details = [
+      'file_local' => __DIR__ . '/../../../fixtures/gff3_loader/gff_duplicate_ids.gff',
+    ];
+
+    
+    $has_exception = false;
+    try {
+      $gff3_importer->create($run_args, $file_details);
+      $gff3_importer->prepareFiles();
+      $gff3_importer->run();
+      $gff3_importer->postRun();
+    } 
+    catch (\Exception $ex) {
+      $message = $ex->getMessage();
+      $has_exception = true;
+    }
+
```

Comment:
We want an assert here to make sure that $has_exception is true when we expect it to be true. That should be outside the catch section.